### PR TITLE
Why Kubernetes

### PR DESCRIPTION
**Pull request by hjacobs**
New blog post "Why Kubernetes?"

### CODE REVIEW

File: posts/why-kubernetes.rst [hunk 1]
```diff
@@ -0,0 +1,115 @@
...
+Nobody needs to hear my opinion, so here it is ;-)
+
+OK, let's imagine you want to run a bunch of containers [#]_, what are your options (sorted alphabetically)?
+
+* `Apache Mesos <http://mesos.apache.org/>`_
+* `AWS ECS <https://aws.amazon.com/ecs/>`_
+* `AWS Elastic Beanstalk <https://aws.amazon.com/elasticbeanstalk/>`_ (OK, let's just `not talk about it <https://twitter.com/QuinnyPig/status/1070848346992963584>`_)
+* `AWS Fargate <https://aws.amazon.com/fargate/>`_ ("serverless containers")
+* ``docker run`` (yes, a very viable option!)
+* `HasiCorp Nomad <https://www.nomadproject.io/>`_
```
Comment (szuecs) [inline posts/why-kubernetes.rst]: HashiCorp

Comment (hjacobs) [reply posts/why-kubernetes.rst]: oh, right, thanks!

File: posts/why-kubernetes.rst [hunk 2]
```diff
@@ -0,0 +1,115 @@
...
+
+Why does it matter that the Kubernetes API is extensible? Having an extensible API matters as you will sooner or later hit an use case not reflected 100% by your infrastructure API,
+and/or you need to integrate with your existing organization's landscape. Kubernetes allows you to extend its API with custom resources (CRDs), e.g. Zalando uses this to `integrate its existing OAuth infrastructure for service-to-service authentication <https://kubernetes-on-aws.readthedocs.io/en/latest/user-guide/zalando-iam.html>`_.
+Custom resources also allow building higher-level abstractions on top of core concepts, e.g. the `Kubernetes StackSet Controller <https://github.com/zalando-incubator/stackset-controller>`_  adds a new (opinionated) StackSet resource to the API for managing application life cycle and traffic switching.
+More common use cases for custom resources are the plentiful `Kubernetes Operators`_. These operators define new CRDs for workloads like `PostgreSQL <https://github.com/zalando/postgres-operator>`_, `etcd <https://github.com/coreos/etcd-operator>`_, `Prometheus <https://github.com/coreos/prometheus-operator>`_, etc.
```
Comment (arjunrn) [inline posts/why-kubernetes.rst]: Depending on when it is opensourced you could also mention the ES operator here.

Comment (hjacobs) [reply posts/why-kubernetes.rst]: Yes, I want to :smile:

**Pull request merged**